### ISSUE TITLE
Request: Mechanism to terminate walks based on Node labels or properties

### ISSUE THREAD

**Issue by gkorland** [FalkorDB/FalkorDB#332]
Created by: @polygonhell
Source: https://github.com/RedisGraph/RedisGraph/issues/2746
Cypher has a straight forwards way to select that only certain edges are walked, but I'd like a mechanism to restrict the walk over nodes, right now I have to specify a where clause to filter paths after the traversal, where what I an trying to do is more naturally terminate the search.

example:  I want to get all the X's connected to a node, regardless of edge type, or intervening nodes in the path, but I want to terminate the search along a specific path on the first X found.

```
 GRAPH.QUERY social "MATCH p = ({id:'01-A-22020-7-126a-702-object-45'})-[*..10]-(s:X) WHERE all(x in tail(reverse(tail(nodes(p)))) WHERE NOT any(l in labels(x) WHERE l = 'X')) RETURN DISTINCT [n in nodes(p) | { l:labels(n), id:n.id } ]"
```

Basically here I have to enumerate all the paths that end in an X even if they traverse an X to get there, then filter out the ones with X's in the middle.
A sublist function would be nice as well.

I'm relatively new to Cypher, so if there is a better way to do this I'm interested.

**Comment by gkorland** [comment]
Origin comment by: @LiorKogan
Wouldn't `MATCH` ... `LIMIT 1` do?

By 'first X found', do you assume that the traversal is executed BFS?

Also, I believe `NOT any` could be replaced with `none`.

**Comment by gkorland** [comment]
Origin comment by: @polygonhell
Limit 1 would only give me the first result, in this case there could be many correct paths, but I want to terminate the search down a specific path once the first is found.
To be clear I want answers that look like 

A -> B -> C-> X
A -> D -> X

But not

A -> B -> C -> X -> E -> X

So the traversal for a path can be terminated if an X is encountered, but judging by the performance/profile command, it's continuing the traversal to the maximum depth and then applying the filter over all of the paths.
